### ISSUE TITLE
Fatfingers in backtesting/hyperopt results

### ISSUE THREAD

**Issue by yazeed** [freqtrade/freqtrade#3088]
Frequently, I seem to come across fat finger candles which seems to mess with backtesting/hyperopt results.

This is a place to discuss potential ways we can address this issue. This also impacts hyperoptimization as it seems to favor configs that pull these noisy fat-finger trades.

I am initially considering a way to identify these fat fingers in backtesting/hyperopt results (extreme outliers?), to ultimately be able to remove them from backtesting/hyperopt results, skip trades that might occur before or after.

![Screen Shot 2020-03-20 at 8 30 36 AM](https://user-images.githubusercontent.com/191483/77144122-535cb280-6a85-11ea-9d81-6d9b2c0ac5b0.png)
![Screen Shot 2020-03-20 at 8 32 47 AM](https://user-images.githubusercontent.com/191483/77144147-63749200-6a85-11ea-971c-a7e9d26471c8.png)

**Comment by xmatthias** [comment]
but why? 
can you remove  / skip them in real trading too? - (spoiler alert: no you can't)

If no, then i don't think there's anything we can (or should) do for this - as this CAN happen in real trading too - and in your case it led to a huge win - but it could lead to a "shakeout" too - where it'd sell for a loss. 
Now you're skipping the loss - and the backtesting-result does not represent the reality anymore.

**Comment by yazeed** [comment]
> but why?
> can you remove / skip them in real trading too? - (spoiler alert: no you can't)

Inn backtesting, it might be fine to keep it as it is, since it obviously made the trade and backtesting is kinda a reality-mirror on a strategy.

But I guess in hyperopt it's more important as it has impact on hyperopt params space discovery.

As you're trying to come up with consistent non-overfitted strategy, hyperopt will continue to pick up configs that favor this fat-finger trade, which is not really desireable as you cannot assume this fat-finger trade will happen again in the future, and this is one major form of unwanted distortion.

I've seen a lot of hyperopt mechanisms where they filter out outlier trades as they are  generally considered noise, and you can do that by looking at the trade roi distribution. On such a chart, this trade would be considered massively unfitting.

.
> 
> If no, then i don't think there's anything we can (or should) do for this - as this CAN happen in real trading too - and in your case it led to a huge win - but it could lead to a "shakeout" too - where it'd sell for a loss.
> Now you're skipping the loss - and the backtesting-result does not represent the reality anymore (same with wins)

**Comment by xmatthias** [comment]
well but how would you approach it? 

removing the trade will falsify the result, as another trade (for another pair) could've happened during that time - and since this fix is not applied to backtesting, hyperopt and backtesting results will not correspond - creating more questions than answers.

I find implementing automatic handling of this a **very** dangerous approach - as it'll cause misalignment between the different commands / modes.

Also, hyperopt should also correctly handle the downside of this (if the finger is pointing down) - it shouldn't go bancrupt just because of that - so leaving that in is the sensible thing to do to me, as it'll have a (hopefully positive) influence on how stoploss is optimized.

**Comment by orehunt** [comment]
- the loss function can already do some noise filtering since it penalizes deviation
- estimators in hyperopt can also deal in general (to a certain degree) with noisy samples, which means their predictions shouldn't be influenced much by particular runs
- you also generally want to cross validate, so the fact that in one instance there an outlier doesn't matter if you split the data

**Comment by xmatthias** [comment]
I'm going to close this as i don't think it would be correct if freqtrade took any action to "normalize" candles before passing them to backtesting.
This would be a dangerous approach - as it would remove reproducability (these candles do happen in real trading, and will cause buys / losses depending on the situation/strategy).

As specified above - this can best be avoided by cross-validation, and also loss-functions for hyperopt may also include some filters on abnormal profits.
